src/libstore/globals.cc

```diff
@@ -156,9 +156,7 @@ StringSet Settings::getDefaultExtraPlatforms()
     // always exec with their own binary preferences.
     if (pathExists("/Library/Apple/System/Library/LaunchDaemons/com.apple.oahd.plist") ||
         pathExists("/System/Library/LaunchDaemons/com.apple.oahd.plist")) {
-        if (std::string{SYSTEM} == "x86_64-darwin")
-            extraPlatforms.insert("aarch64-darwin");
-        else if (std::string{SYSTEM} == "aarch64-darwin")
+        if (std::string{SYSTEM} == "aarch64-darwin")
```

Comment:
Hmm this seems to still be wrong, and also only assign this for aarch64-darwin now?

Reply:
An aarch64-darwin with Rosetta indeed does have x86_64-darwin as a supported (extra) platform.

Reply:
Nix doesn't need the current system to be an "extra" platform :).

Reply:
I did a brief test with an M1 and Intel Mac I have access to, and the `/System/Library/LaunchDaemons/com.apple.oahd.plist` path existed on both (including the Intel Mac), which I think is why the "if x86, add aarch" branch was removed -- it ends up activating on non-Rosetta machines.